lib/web/apiserver.go

```diff
@@ -108,6 +109,9 @@ import (
 	"github.com/gravitational/teleport/lib/web/ui"
 )
 
+// apiPrefixRegex matches pathnames starting with /v<version num>/<any characters>
+var apiPrefixRegex = regexp.MustCompile(`^/v(\d+)/(.+)`)
+
```

Comment:
I ended up going with this approach to keep the old behavior the same. I could've done something stricter where regex is like `^/v(\d+)/(webapi|enterprise|scripts|.well-known|workload-identity)` but once you introduce another non-prefix leading path, you have to remember to update this regex.

For cases where route is double prefixed like this `/v1/v2/webapi`, I do another regex match on the path after the first prefix to prevent that.

I couldn't see other issues with this approach but let me know.

Reply:
If identifying version prefix is all that we are interested in, using regexp is overkill imho. Any reason using regexp over string match? We could split the path string and check for `v1`, `v2`?

Reply:
i may be just missing misunderstanding, but if we did string matches, wouldn't it just grow over time? like lets say we introduce `v3`, we have to now include a `... || strings.HasPrefix("v3")`

or do you think this is okay because it's highly unlikely we'll have like... 10 versions?

Reply:
actually i think i'm overthinking this. we only need to check for `v1` and if `v1` strip it off. all other prefixes will be required from v1+. this was how the original code worked without me touching anything.

the problem was, this also became valid `v1/v2/webapi...`, so i can first do a string match for `v1`, then do a regex check for the remaining path, is that better?

Reply:
Right, the only thing you need here is check if the first version prefix is `v1` and then strip it off if found. If you want to prevent `v1/v2` footgun, we can just whitelist known second path `(webapi|enterprise|scripts|.well-known|workload-identity)` which adds a bonus of documentation of supported paths. I think that will still be better than using regexp here.

Reply:
how about this: https://github.com/gravitational/teleport/pull/50472/commits/bc43b78751ce769f5cdd30d65f696aa6013d6ca7